### PR TITLE
#135 Revision should not be added to the recipe

### LINKED ISSUE #135: [conan center] check there are no scm and revision mode attributes

### PR DESCRIPTION
**Pull request by uilianries**
Both `scm` and `revision_mode` should not used in the recipes

closes #135 

WIKI:

#### **<a name="KB-H039">#KB-H039</a>: "NOT ALLOWED ATTRIBUTES"**

Some attributes are experimental features or may influence the package id, so that they should be avoided.

### CODE REVIEW

**Comment by danimtb** [comment]
I think the message "NO REVISION" is a bit unclear to me. I would make the message more generic like "NOT ALLOWED ATTRIBUTES" (we need a better name) and indicate the attribute that is not allowed.

File: hooks/conan-center.py [hunk 1]
```diff
@@ -345,6 +346,18 @@ def test(out):
...
+    @run_test("KB-H039", output)
+    def test(out):
+        forbidden_attrs = []
+        if getattr(conanfile, "scm", None):
+            forbidden_attrs.append("scm")
+        if re.search(r"revision_mode\s=", conanfile_content):
+            forbidden_attrs.append("revision_mode")
+
+        if forbidden_attrs:
+            out.error("Conanfile should not contain attributes related to revision. "
```
Comment (danimtb) [inline hooks/conan-center.py]: please change the error message

Comment (uilianries) [reply hooks/conan-center.py]: done

File: hooks/conan-center.py [hunk 2]
```diff
@@ -345,6 +346,18 @@ def test(out):
...
+    @run_test("KB-H039", output)
+    def test(out):
+        forbidden_attrs = []
+        if getattr(conanfile, "scm", None):
+            forbidden_attrs.append("scm")
+        if re.search(r"revision_mode\s=", conanfile_content):
+            forbidden_attrs.append("revision_mode")
+
+        if forbidden_attrs:
+            out.error("Conanfile should not contain the follow attributes: '{}'. Remove them."
```
Comment (danimtb) [inline hooks/conan-center.py]:
```suggestion
            out.error("Conanfile should not contain attributes: '{}'"
```

Comment (uilianries) [reply hooks/conan-center.py]: @danimtb Done!

**Review by danimtb — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**